docs/src/starknet/verify.md

```diff
@@ -0,0 +1,43 @@
+# Verifying Contracts
+
+## Overview
+
+Starknet Foundry `sncast` supports verifying Cairo contract classes with the `sncast verify` command by submitting the source code to a selected verification provider. Verification provides transparency, making the code accessible to users and aiding debugging tools.
+
+The verification provider guarantees that the submitted source code aligns with the deployed contract class on the network by compiling the source code into Sierra bytecode and comparing it with the network-deployed Sierra bytecode.
+
+For detailed CLI description, see [verify command reference](../appendix/sncast/verify.md).
+
+> ⚠️ **Warning**
+> Please be aware that submitting the source code means it will be publicly exposed through the provider's APIs.
+
+## Verification Providers
+
+### Walnut
+
+Walnut is a tool for step-by-step debugging of Starknet transactions. You can learn more about Walnut here [walnut.dev](https://walnut.dev). Note that Walnut requires you to specify the Starknet version in your `Scarb.toml` config file.
+
+## Example
+
+First, ensure that you have created a `Scarb.toml` file for your contract (it should be present in the project directory or one of its parent directories). Make sure the contract has already been deployed on the network.
+
+Then run:
+
+```shell
+$ sncast --url http://127.0.0.1:5050/rpc \
+    verify \
+    --contract-address 0x8448a68b5ea1affc45e3fd4b8b480ea36a51dc34e337a16d2567d32d0c6f8b \
+    --contract-name SimpleBalance \
+    --verifier walnut \
+    --network mainnet
+
+You are about to submit the entire workspace's code to the third-party chosen verifier at walnut, and the code will be publicly available through walnut's APIs. Are you sure? (Y/n) Y
+
+command: verify
+message: Contract successfully verified
+```
+
+After verification with Walnut verifier, the contract can be debugged using Walnut's transaction step-by-step debugger.
```

Comment:
If this is a general section (see the title marking that you did at `Example`), this does not belong here 

Also i don't think this is relevant at all

Reply:
It still lacks the explanation where should i look for the confirmation, the "green check mark" that i'm after when doing the verification etc.

Is there a URL that i can use?